### PR TITLE
Delay Shop Scroll List

### PR DESCRIPTION
**Pull request by anethus**
Throttling events for scrolling in vendor shop list when user hold down mouse on a scroll buttons

### CODE REVIEW

File: src/Game/UI/Gumps/ShopGump.cs [hunk 1]
```diff
@@ -447,6 +439,30 @@ public override bool Draw(UltimaBatcher2D batcher, int x, int y)
             return base.Draw(batcher, x, y);
         }
 
+        private void ProcessListScroll()
+        {
+            if ((DateTime.Now - _lastMouseEventTime).TotalMilliseconds >= SCROLL_DELAY)
```
Comment (andreakarasho) [inline src/Game/UI/Gumps/ShopGump.cs]: repleace `DateTime.Now` with the `Time.Ticks`
The `DateTime` struct is not necessary here and creates to much overheads under the hood

File: src/Game/UI/Gumps/ShopGump.cs [hunk 2]
```diff
@@ -447,6 +439,30 @@ public override bool Draw(UltimaBatcher2D batcher, int x, int y)
...
+                        _shopScrollArea.Scroll(false);
+                        break;
+                    case ButtonScroll.RightScrollUp:
+                        _transactionScrollArea.Scroll(true);
+                        break;
+                    case ButtonScroll.RightScrollDown:
+                        _transactionScrollArea.Scroll(false);
+                        break;
+                }
+                _lastMouseEventTime = DateTime.Now;
```
Comment (andreakarasho) [inline src/Game/UI/Gumps/ShopGump.cs]: RE: `DateTime` stuff

File: src/Game/UI/Gumps/ShopGump.cs [hunk 3]
```diff
@@ -57,6 +57,9 @@ enum ButtonScroll
             RightScrollUp,
             RightScrollDown
         }
+        // Scroll Delay (in ms)
+        private const int SCROLL_DELAY = 60;
+        private DateTime _lastMouseEventTime = DateTime.Now;
```
Comment (andreakarasho) [inline src/Game/UI/Gumps/ShopGump.cs]: RE: `DateTime` stuff

File: src/Game/UI/Gumps/ShopGump.cs (hunk 1 above)
Comment (anethus) [reply src/Game/UI/Gumps/ShopGump.cs]: done

**Pull request merged**